lib/elixir/lib/uri.ex

```diff
@@ -88,7 +88,22 @@ defmodule URI do
 
   """
   @spec encode_query(term) :: binary
-  def encode_query(l), do: Enum.map_join(l, "&", &pair/1)
+  def encode_query(enumerable) do
+    Enum.map_join(enumerable, "&", &encode_kv_pair/1)
+  end
+
+  defp encode_kv_pair({key, _}) when is_list(key) do
+    raise ArgumentError, "encode_query/1 keys cannot be lists, got: #{inspect key}"
+  end
+
+  defp encode_kv_pair({_, value}) when is_list(value) do
+    raise ArgumentError, "encode_query/1 values cannot be lists, got: #{inspect value}"
+  end
+
+  defp encode_kv_pair({key, value}) do
+    encode_www_form(Kernel.to_string(key)) <>
+    "=" <> encode_www_form(Kernel.to_string(value))
```

Comment:
@lexmag I left this unchanged on purpose because I don't think we have a rule for that, do we? If we're consistent across all Elixir, then I'll be happy to change it of course :)